core/src/client.rs

```diff
@@ -323,10 +323,10 @@ impl APIClient {
         }
 
         let resp: QueryResponse = resp.json().await?;
+        self.handle_session(&resp.session).await;
```

Comment:
I guess this will get the error `query error: code: 4002, message: current transaction is aborted, commands ignored until end of transaction block` as well after `commit` failed. Because failed session state will be updated into the client.

Reply:
session state will be autocommit

Reply:
There would be only a `error` field in response if the query fails?

Reply:
> session state will be autocommit

@youngsofun  Now the session state will be `Fail` if commit failed. I modify this line and pass the tests https://github.com/datafuselabs/databend-go/pull/104/files#diff-6b6b5b4573a4b05f56ba2da3407d181f157943895ec000137734b382cadecc17R337

Reply:
@hantmac fixed